src/test/java/org/influxdb/InfluxDBLogLevelTest.java

```diff
@@ -0,0 +1,50 @@
+package org.influxdb;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import org.influxdb.dto.Pong;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+/**
+ * Test the InfluxDBimpl log level setting from system property.
+ *
+ * @author hoan.le [at] bonitoo.io
+ *
+ */
+@RunWith(JUnitPlatform.class)
+public class InfluxDBLogLevelTest {
+  @Test
+  public void testLogLevelProperties() {
+    
+    String oldLogLevel = System.getProperty(InfluxDB.LOG_LEVEL_PROP);
+    List<String> logLevels = Arrays.asList(null, "NONE", "BASIC", "HEADERS", "FULL", "abc");
+    logLevels.forEach(logLevel -> {
+      System.out.println("LogLevel = " + logLevel);
+      Optional.ofNullable(logLevel).ifPresent(value -> {
+        System.setProperty(InfluxDB.LOG_LEVEL_PROP, value);
```

Comment:
Hi,

I went through the code with the following findings:

1) InfluxDB.LOG_LEVEL_PROP ? It should be InfluxDB.LOG_LEVEL_PROPERTY. It is 5 chars more only.
2) I don't even like the 'initLogLevel' construct. Rather we should do:

   `  this.setLogLevel(this.getConfiguredLogLevel())`

getConfiguredLogLevel() would just read the configuration property and set a default value for the LogLevel. Like this, we have two functions where it is clear what is happening here and there this one function less that has side effects.

Vlada

Reply:
@dubsky , thank you Vlada,
I will fix as @fmachado requested, for me it's not too bad